src/logreader.c

```diff
@@ -168,19 +166,16 @@ sparkey_returncode sparkey_logiter_create(sparkey_logiter **iter_ref, sparkey_lo
   iter->block_len = 0;
   iter->state = SPARKEY_ITER_NEW;
 
-  switch (log->header.compression_type) {
-  case SPARKEY_COMPRESSION_NONE:
+  if (log->header.compression_type == SPARKEY_COMPRESSION_NONE) {
     iter->compression_buf_allocated = 0;
-    break;
-  case SPARKEY_COMPRESSION_SNAPPY:
+  } else if (sparkey_uses_compressor(log->header.compression_type)) {
```

Comment:
Use this as the primary if-condition, remove the one on line 169 and remove the error case block on line 178